ADAL/src/public/ADAuthenticationContext.h

```diff
@@ -455,6 +455,24 @@ typedef enum
                           forceRefresh:(BOOL)forceRefresh
                        completionBlock:(ADAuthenticationCallback)completionBlock;
 
+
+/*! Follows the OAuth2 protocol (RFC 6749). The function accepts claims challenge returned from middle tier service, which will be sent to token endpoint. If claims parameter is not nil/empty, access tokens in cache will be skipped and refresh token will be tried.
+ If reauthorization is needed, the method will return an error with code AD_ERROR_USER_INPUT_NEEDED.
+ @param resource The resource whose token is needed.
+ @param clientId The client identifier
+ @param redirectUri The redirect URI according to OAuth2 protocol
+ @param userId The user to be used to look up the access token and refresh token in cache
+ @param claims The claims parameter that needs to be sent to the token endpoint. It should be URL-encoded.
+ @param completionBlock The block to execute upon completion. You can use embedded block, e.g. "^(ADAuthenticationResult res){ <your logic here> }"
+ */
+- (void)acquireTokenSilentWithResource:(NSString *)resource
+                              clientId:(NSString *)clientId
+                           redirectUri:(NSURL *)redirectUri
+                                userId:(NSString *)userId
+                                claims:(NSString *)claims
+                       completionBlock:(ADAuthenticationCallback)completionBlock;
+
+
```

Comment:
should we update the comments of `[ADAuthenticationContext acquireTokenWithResource:...]`? as the logic has changed to skip access token rather than skipping the whole cache :)

Reply:
Good point, will update :)